File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/interactive/InteractiveElementSkill.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait InteractiveElementSkill extends TypeDeserializer {
  val typeId = 219

  protected trait Type extends BaseType with Serializable {
    val typeId = InteractiveElementSkill.typeId

    val skillId: Int
    val skillInstanceUid: Int

    def serialize(buf: Buffer) {
      buf.writeInt(skillId)
      buf.writeInt(skillInstanceUid)
    }

  }

  protected def create(skillId: Int, skillInstanceUid: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val skillId = buf.readInt()
    val skillInstanceUid = buf.readInt()

    create(skillId, skillInstanceUid)
  }

}

object InteractiveElementSkill extends InteractiveElementSkill {
  case class TypeImpl(skillId: Int, skillInstanceUid: Int) extends Type
  override def create(skillId: Int, skillInstanceUid: Int) = TypeImpl(skillId, skillInstanceUid)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(skillId: Int, skillInstanceUid: Int) = create(skillId, skillInstanceUid)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/InteractiveElementSkill.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InteractiveElementSkill extends NetworkType {
    public static final short TYPE_ID = 219;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int skillId;
    public int skillInstanceUid;
    
    public InteractiveElementSkill() { }
    
    public InteractiveElementSkill(int skillId, int skillInstanceUid) {
        this.skillId = skillId;
        this.skillInstanceUid = skillInstanceUid;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(skillId);
        buf.writeInt(skillInstanceUid);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        skillId = buf.readInt();
        if (skillId < 0)
            throw new RuntimeException("Forbidden value on skillId = " + skillId + ", it doesn't respect the following condition : skillId < 0");
        skillInstanceUid = buf.readInt();
        if (skillInstanceUid < 0)
            throw new RuntimeException("Forbidden value on skillInstanceUid = " + skillInstanceUid + ", it doesn't respect the following condition : skillInstanceUid < 0");
    }
    
}
